docs/concepts/demisto-sdk.md

```diff
@@ -0,0 +1,389 @@
+
+#### Examples and Use Cases:
+ - Validate all committed files:
+    ```buildoutcfg
+    demisto-sdk validate
+    ```
+   This command will identify all the files that were **committed** via `git` in your current branch and will run validations on them.
+   Do note that this is the variation of the command that runs in our build process, in case of a build failure run this to recreate it locally.
+ 
+ 
+ - Validate all committed and staged files:
+    ```buildoutcfg
+    demisto-sdk validate -g
+    ```
+   This command will identify all the files that were **committed and staged** via `git` in your current branch and will run validations on them.
+ 
+ 
+ - Validate a specific file:
+    ```buildoutcfg
+    demisto-sdk validate -i Packs/myPack/Integrations/myIntegration/myIntegration.yml
+    ```
+   This will run validation only on the file `Packs/myPack/Integrations/myIntegration/myIntegration.yml`. This variation does not detect the file status via `git` and thus will not check for backwards comparability changes.
+  
+  
+ - Validate all committed and staged files, print ignored files and errors and debug git:
+    ```buildoutcfg
+    demisto-sdk validate -g --print-ignored-errors --print-ignored-files --debug-git
+    ```
+   This command will validate all committed and staged files and will print out some additional information which is hidden by default: 
+    - Which files were ignored.
+    - Error codes that were ignored.
+    - A report on which files where identified and in what status via `git`.
+   This additional information might be helpful in debugging SDK commands or trying to figure out why a file was not validate of some reason.
+
+#### Validate Error Structure and Ignoring Errors
+Each one of our validation errors follows a basic structure to ease it's readability:
+```buildoutcfg
+File_Path: [Error_Code] - Error Message
+```
+
+For example:
+```buildoutcfg
+Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml: [BA100] - The version for our files should always be -1, please update the file.
+```
+
+At the end of the validation a summery would appear with all the file paths and error codes that were encountered.
+
+The error codes serve two main functions:
+1. The first two letters are used to identify the type of the error encountered for example: `DO` is a Docker related error, `BC` is backwards compatibility related and `ST` is a yml/json structure error.
+    A full list of abbreviations can be found in the full validate documentation linked below.
+2. The error code is used to ignore errors. To ignore an error go to the file's `.pack-ignore` file and input the following structure:
+   ```buildoutcfg
+    [file:file_name]
+    ignore=<error_code_to_ignore>
+    ```
+   For example:
+   ```buildoutcfg
+    [file:CortexXDRIR.yml]
+    ignore=IN126,IN135
+    ```
+   * Note: **not all error codes are ignorable! Also, it is always preferable to fix the error rather than trying to ignore it!** 
```

Comment:
```suggestion
   * Note: **Not all error codes are ignorable! Also, it is always preferable to fix the error rather than trying to ignore it!** 
```